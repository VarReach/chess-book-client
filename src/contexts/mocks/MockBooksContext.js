import React, { Component } from 'react'

const MockBooksContext = React.createContext({
  bookId: null,
  books: {},
});

export default MockBooksContext;

export class MockBooksProvider extends Component {
  constructor(props) {
    super(props);
    this.state = {
      bookId: 1,
      books: {
        '1': {
          title: 'Testing',
          blurb: 'Testing',
          default_book: true,
          id: 1,
          chapter_order: null,
          chapters: null,
        }
      },
    };
  }

  render() {
    return (
      <MockBooksContext.Provider value={this.state}>
        {this.props.children}
      </MockBooksContext.Provider>
    )
  }
}
