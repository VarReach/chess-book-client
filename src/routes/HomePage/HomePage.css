.home-page__header {
  display: flex;
  align-items: center;
  position: relative;
  margin-top: 125px !important;
  margin-bottom: 140px !important;
  padding: 0 30px !important;
  color: #242425;
}

.home-page__book-details {
  width: calc(100% - 35px);
  position: relative;
}

.home-page__header h1, .home-page__header h3 {
  padding-right: 5px;
  margin: 0;
}

.home-page__header-book-button {
  position: relative;
  padding: 0;
  padding-top: 3px;
  width: 40px;
  height: 40px;
  background-color: transparent;
  border: none;
  color: #838383;
  font-size: 22px;
  outline: none;
  background-color: #f7f7f7;
  border: 1px solid #c0c0c0;
  box-shadow: 0 3px 0 0 #d9d9d9;
}

.home-page__header-book-button:hover {
  cursor: pointer;
}

.home-page__header-book-button:hover, .home-page__header-book-button:focus {
  box-shadow: none;
  outline: 3px solid #d9d9d9;
}

.book-select__dropdown-menu {
  width: 300px;
  height: 190px;
  right: 90px;
}

.book-select__dropdown-menu:after {
	left: 100%;
	top: calc(50% - 16px);
	border: solid transparent;
	content: " ";
	height: 0;
	width: 0;
	position: absolute;
	pointer-events: none;
	border-color: #00000000;
  border-left-color: #242831;
  /* border-width: 0 16px 8px 16px; */
  border-width: 16px 0 16px 8px;
}

@media only screen and (max-width: 600px) {
  .home-page__header {
    margin-top: 60px !important;
    margin-bottom: 80px !important;
  }

  .home-page__header h1 {
    font-size: 32px;
  }

  .home-page__header h3 {
    font-size: 20px;
  }

  .book-select__dropdown-menu {
    top: 90px;
    right: 30px;
  }

  .book-select__dropdown-menu:after {
    bottom: 100%;
    top: unset;
    left: unset;
    right: 5px;
    border-color: #00000000;
    border-bottom-color: #242831;
    /* border-width: 0 16px 8px 16px; */
    border-width: 0 16px 8px 16px;
  }
}


@media only screen and (max-width: 400px) {
  .home-page__header h1 {
    font-size: 26px;
  }

  .home-page__header h3 {
    font-size: 16px;
  }
}