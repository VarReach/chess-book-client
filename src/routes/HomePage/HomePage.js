import React, { Component } from 'react'
import BooksContext from '../../contexts/BooksContext';
import BooksApiService from '../../services/books-api-service';
import ChapterList from '../../components/ChapterList/ChapterList';
import BookSelectMenu from '../../components/BookSelectMenu/BookSelectMenu';
import './HomePage.css';

class ChaptersList extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showDropDown: false,
    };
    this.dropDown = React.createRef();
  }

  static contextType = BooksContext;

  componentDidMount() {
    BooksApiService.getAllBooks()
      .then(books => {
        let defaultBook;
        let booksObj = {};
        books.forEach(book => {
          booksObj[book.id] = book;
          book.default_book && (defaultBook = book.id);
        });
        this.context.setBooks(booksObj, defaultBook);
      })
      .catch(this.context.setError);
  }

  toggleDropDown = (e) => {
    if (e && this.dropDown.current.contains(e.target)) {
      return;
    }
    this.state.showDropDown ? this.hideDropDown() : this.showDropDown();
  }

  showDropDown = () => {
    this.setState({ showDropDown: true }, () => {document.addEventListener('click', this.hideDropDown)});
  }

  hideDropDown = () => {
    this.setState({ showDropDown: false }, () => {document.removeEventListener('click', this.hideDropDown)});
  }

  render() {
    const bookId = this.context.bookId;
    const chapters = (bookId && this.context.books[this.context.bookId].chapters) || [];
    return (
      <>
        <header role="banner" className="container home-page__header">
          <div className="home-page__book-details">
            {bookId && <h1>{this.context.books[bookId].title}</h1>}
            {bookId && <h3>{this.context.books[bookId].blurb}</h3>}
          </div>
          <button className="home-page__header-book-button" onClick={this.toggleDropDown}><i className="fas fa-bars"/></button>
          <div className={"dropdown book-select__dropdown-menu" + (!this.state.showDropDown ? ' hidden' : '')} ref={this.dropDown}>
            <BookSelectMenu />
          </div>
        </header>
        <section className="container">
          <ChapterList chapters={chapters}/>
        </section>
      </>
    );
  }
}

export default ChaptersList
