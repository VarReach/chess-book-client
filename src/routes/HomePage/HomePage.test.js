import React from 'react';
import ReactDOM from 'react-dom';
import HomePage from './HomePage';
import { MockBooksProvider } from '../../contexts/mocks/MockBooksContext';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('HomePage tests', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <HomePage />,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders given context', () => {
    const wrapper = shallow(
      <MockBooksProvider>
        <HomePage />
      </MockBooksProvider>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
