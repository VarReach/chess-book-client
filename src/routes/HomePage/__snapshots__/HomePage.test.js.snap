// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage tests renders given context 1`] = `
<ContextProvider
  value={
    Object {
      "bookId": 1,
      "books": Object {
        "1": Object {
          "blurb": "Testing",
          "chapter_order": null,
          "chapters": null,
          "default_book": true,
          "id": 1,
          "title": "Testing",
        },
      },
    }
  }
>
  <ChaptersList />
</ContextProvider>
`;
